hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/entity/TopicBuilderImpl.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) 2022-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.consensus.impl.entity;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.node.app.service.consensus.entity.Topic;
+import com.hedera.node.app.service.consensus.entity.TopicBuilder;
+import com.hedera.node.app.service.token.entity.Account;
+import com.hedera.node.app.spi.key.HederaKey;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * An implementation of {@link TopicBuilder} for building Topic instances. This class is
+ * <strong>not</strong> exported from the module.
+ * module
+ */
+public class TopicBuilderImpl implements TopicBuilder {
```

Comment:
We should discuss if we want to move builders & record (plain data objects) directly in the API layer.

Reply:
Yes, in my next PR I will move `Account` And `AccountBuilder` to SPI. I am not sure if all entities need to be moved though. Agreed, I will bring this up in next meeting. Thanks for pointing out @hendrikebbers